libs/designer/src/lib/ui/templates/__tests__/templateCard.spec.tsx

```diff
@@ -0,0 +1,110 @@
+import type { AppStore } from '../../../core/state/templates/store';
+import { setupStore, type RootState } from '../../../core/state/templates/store';
+import type { Template } from '@microsoft/logic-apps-shared';
+import { renderWithProviders } from '../../../__test__/template-test-utils';
+import { screen } from '@testing-library/react';
+import { TemplateCard } from '../cards/templateCard';
+
+describe('ui/templates/templatesDesigner', () => {
+  let store: AppStore;
+  let minimalStoreData: Partial<RootState>;
+  let template1Manifest: Template.Manifest;
+  let template2Manifest: Template.Manifest;
+
+  beforeAll(() => {
+    template1Manifest = {
+      title: 'Template 1',
+      description: 'Template 1 Description',
+      skus: ['standard'],
+      kinds: ['stateful'],
+      artifacts: [
+        {
+          type: 'workflow',
+          file: 'workflow.json',
+        },
+        {
+          type: 'description',
+          file: 'description.md',
+        },
+      ],
+      connections: [
+        {
+          id: 'connection example 1',
+        },
+        {
+          id: 'connection example 2',
+        },
+      ],
+      parameters: [],
+    };
+    template2Manifest = {
+      title: 'Template 2',
+      description: 'Template 2 Description',
+      skus: ['standard', 'consumption'],
+      kinds: ['stateful', 'stateless'],
+      artifacts: [
+        {
+          type: 'workflow',
+          file: 'workflow.json',
+        },
+        {
+          type: 'description',
+          file: 'description.md',
+        },
+      ],
+      connections: [
+        {
+          id: 'connection example 1',
+        },
+      ],
+      parameters: [],
+    };
+  });
+
+  it('Renders TemplateCard and loads template state correctly on buttons click', async () => {
+    minimalStoreData = {
+      manifest: {
+        availableTemplateNames: ['template1', 'template2'],
+        availableTemplates: {
+          template1: template1Manifest,
+          template2: template2Manifest,
+        },
+      },
+    };
+    store = setupStore(minimalStoreData);
+
+    renderWithProviders(<TemplateCard templateName="template1" />, { store });
+
+    expect(screen.getByText('Create Workflow')).toBeDefined();
+    expect(screen.getByText('Quick View')).toBeDefined();
+    screen.getByText('Create Workflow').click();
+    expect(store.getState().template.templateName).toBe('template1');
+  });
+
+  it('Renders TemplateCard and Opens the right panel', async () => {
+    minimalStoreData = {
+      manifest: {
+        availableTemplateNames: ['template1', 'template2'],
+        availableTemplates: {
+          template1: template1Manifest,
+          template2: template2Manifest,
+        },
+      },
+    };
+    store = setupStore(minimalStoreData);
+
+    renderWithProviders(<TemplateCard templateName="template2" />, { store });
+
+    expect(screen.getByText('Create Workflow')).toBeDefined();
```

Comment:
I wonder how this will work for later implementations if we support intl

Reply:
TemplateCard already has a intl 'Create Workflow' text for the button, and the test passed!